pkg/operator/controller/canary/networkpolicy_test.go

```diff
@@ -0,0 +1,80 @@
+package canary
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test_desiredCanaryNetworkPolicy verifies the desired NetworkPolicy for canary pods.
+func Test_desiredCanaryNetworkPolicy(t *testing.T) {
+	np := desiredCanaryNetworkPolicy()
+
+	if np.Name != "ingress-canary" || np.Namespace != "openshift-ingress-canary" {
+		t.Fatalf("unexpected name/namespace: %s/%s", np.Namespace, np.Name)
+	}
+	if np.Labels["ingress.openshift.io/canary"] != canaryControllerName {
+		t.Fatalf("missing or wrong owner label: %#v", np.Labels)
+	}
+	// Ensure pod selector is set via helper.
+	if np.Spec.PodSelector.MatchLabels == nil {
+		t.Fatalf("pod selector should be set")
+	}
+	// Ensure ingress rules include ports 8888 and 8443.
+	want := map[int32]struct{}{8888: {}, 8443: {}}
+	got := map[int32]struct{}{}
+	for _, r := range np.Spec.Ingress {
+		for _, p := range r.Ports {
+			if p.Protocol != nil && *p.Protocol == corev1.ProtocolTCP && p.Port != nil && p.Port.IntVal != 0 {
+				got[p.Port.IntVal] = struct{}{}
```

Comment:
`struct{}` is passé.  Consider using `"k8s.io/apimachinery/pkg/util/sets".Set[int32]`.